OpenRA.Game/Primitives/PriorityQueue.cs

```diff
@@ -13,15 +13,28 @@
 
 namespace OpenRA.Primitives
 {
-	public class PriorityQueue<T>
-			where T : IComparable<T>
+	public interface IPriorityQueue<T>
 	{
-		List<T[]> items = new List<T[]>();
+		void Add(T item);
+		bool Empty { get; }
+		T Peek();
+		T Pop();
+	}
+
+	public class PriorityQueue<T> : IPriorityQueue<T>
+	{
+		readonly IList<T[]> items;
```

Comment:
Since this is private you should keep it typed as a concrete ```List<T[]>```. That will avoid any overhead of using the interface. (There probably isn't much here, but it's not like the interface is really doing much for us either).